server/scripts/sendEmailBlast.js

```diff
@@ -68,117 +77,160 @@ function createPayload(recipient, uuid) {
-  const batchTime = (new Date() - now) / 1000;
-  console.log(
-    `Sent ${FgGreen}${start}${Reset} to ${FgGreen}${
-      start + batch
-    }${Reset}: ${FgGreen}${batchTime.toFixed(2)}${Reset}s, ${FgGreen}${(batch / batchTime).toFixed(
-      1
-    )}${Reset}/s`
-  );
-  return blastRecursive(start + batch, end, max, batch, chesConfiguration);
-}
-
-async function blast(start, end, batch) {
-  const { count } = (await countEmails())[0];
-  console.log(`Blasting ${FgGreen}${count}${Reset} emails...`);
-  await updateToken();
+  if (errors) {
+    console.log(`${colours.fg.red}${errors} error${errors > 0 ? 's' : ''} logged.${colours.reset}`);
+  } else {
+    console.log(`${colours.fg.green}No errors encountered.${colours.reset}`);
+  }
+};
 
-  const now = new Date();
-  await blastRecursive(start, end, count, batch, config);
-  const totalTime = (new Date() - now) / 1000;
-  console.log(
-    `Sent ${FgGreen}${count}${Reset} emails in ${totalTime.toFixed(2)}s at a rate of ${(
-      count / totalTime
-    ).toFixed(2)}`
+function askQuestion(query, defaultValue) {
+  const rl = readline.createInterface({
+    input: process.stdin,
+    output: process.stdout,
+  });
+
+  return new Promise((resolve) =>
+    rl.question(`${query} [${defaultValue}]: `, (ans) => {
+      rl.close();
+      resolve(ans ? parseInt(ans, 10) : defaultValue);
+    })
   );
 }
 
-const writeFailedSends = (failures) => {
-  console.log(`There were ${failures.length} failed sends.`);
-  if (failures.length === 0) return;
-  const timestamp = new Date().toJSON();
-  const failureJson = JSON.stringify(failures);
-  console.log(failures);
-  writeFileSync(path.join(__dirname, `failed_ches_sends-${timestamp}.json`), failureJson);
-};
-
 (async function emailBlast() {
-  const arglength = process.argv.length;
-  const mode = arglength > 2 ? process.argv[2] : 'default';
-  const start = arglength > 3 ? parseInt(process.argv[3], 10) : 0;
-  const end = arglength > 4 ? parseInt(process.argv[4], 10) : -1;
-  const batch = arglength > 5 ? parseInt(process.argv[5], 10) : 100;
-  switch (mode) {
-    case 'all':
-      await blast(0, -1, 100);
-      writeFailedSends(failedSends);
-      break;
-    case 'index':
-      console.log(start, end, batch);
-      await blast(start, end, batch);
-      writeFailedSends(failedSends);
-      break;
-    case 'count':
-      console.log((await countEmails())[0].count);
-      break;
-    default:
-      console.log('No action specified');
+  const mailRate = await askQuestion('Max Rate', 15);
+  const retryDelay = await askQuestion('Retry Delay (ms)', 10000);
+  const retryLimit = await askQuestion('Retry Limit', 10);
+  const mailLimit = await askQuestion('Total Mail Limit', 10000);
+
+  /**
+   * CHES Daily Limit = 10,000
+   * https://github.com/bcgov/common-hosted-email-service/wiki/Best-Practices#rate-limits
+   */
+  if (mailLimit > 10000) {
+    console.log(`${colours.fg.red}Defaulting to 10,000 emails.${colours.reset}`);
```

Comment:
I don't think this is actually true.

Reply:
Definitely not true as a hard limit, this is in reference to a limit set by PM